src/mumble/Log.cpp

```diff
@@ -803,13 +805,32 @@ void Log::log(MsgType mt, const QString &console, const QString &terse, bool own
 		if (!(Global::get().mw->isActiveWindow() && Global::get().mw->qdwLog->isVisible())) {
 			// Message notification with window highlight
 			if (flags & Settings::LogHighlight) {
-				QApplication::alert(Global::get().mw);
+				Global::get().mw->highlightWindow();
 			}
 
 			// Message notification with balloon tooltips
 			if (flags & Settings::LogBalloon) {
 				// Replace any instances of a "Object Replacement Character" from QTextDocumentFragment::toPlainText
-				postNotification(mt, plain.replace("\xEF\xBF\xBC", tr("[embedded content]")));
+				plain = plain.replace("\xEF\xBF\xBC", tr("[embedded content]"));
+
+				QSystemTrayIcon::MessageIcon msgIcon;
+				switch (mt) {
+					case DebugInfo:
+					case CriticalError:
+						msgIcon = QSystemTrayIcon::Critical;
+						break;
+					case Warning:
+						msgIcon = QSystemTrayIcon::Warning;
+						break;
+					case TextMessage:
+					case PrivateTextMessage:
+						msgIcon = QSystemTrayIcon::NoIcon;
+						break;
+					default:
+						msgIcon = QSystemTrayIcon::Information;
+						break;
```

Comment:
I would initialize `msgIcon` with this value and drop the default clause. I usually try to never use default clauses for enum values in order to get the compiler to yell at me if a new enum value is introduced that is not yet handled.

Reply:
Ok. But we can skip the declaration+initialization because we have to handle all remaining like 20 enum cases then anyway.